### ISSUE TITLE
10.0.7 : Error: Incorrect integer value: '' for column `glpi`.`glpi_agents`.`items_id` at row 1

### ISSUE THREAD

**Issue by steph461** [glpi-project/glpi#15036]
### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Version

10.0.7

### Bug description

Hello,

When I Want change name of agents after to have save, i have one error message SQL in the log
![image](https://github.com/glpi-project/glpi/assets/68054991/089dea1f-7a22-435f-9d31-90c3375c4950)
![image](https://github.com/glpi-project/glpi/assets/68054991/e3f8e8cc-0be5-4461-8182-a1baa54b00e9)

### Relevant log output

```shell
[2023-06-28 14:44:48] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 370
  *** MySQL query error:
  SQL: UPDATE `glpi_agents` SET `name` = 'xxxxxxxxxxxxxxxxxxxxxxx', `items_id` = '' WHERE `id` = '343'
  Error: Incorrect integer value: '' for column `glpi`.`glpi_agents`.`items_id` at row 1
  Backtrace :
  src/DBmysql.php:1439                               
  src/CommonDBTM.php:686                             DBmysql->update()
  src/CommonDBTM.php:1672                            CommonDBTM->updateInDB()
  front/agent.form.php:96                            CommonDBTM->update()
  {"user":"7@srv-glpi"}
```

### Page URL

/glpi/front/agent.form.php?id=XX

### Steps To reproduce

1. go to the computer
2. Select Agent
![image](https://github.com/glpi-project/glpi/assets/68054991/f4ceaf76-56e9-4c56-afa2-da98c826591f)
3. Change Name
4. Save

### Your GLPI setup information

<details><summary>Informations sur le système, l'installation et la configuration</summary><pre>GLPI 10.0.7 (/glpi => /var/www/html/glpi)
Installation mode: TARBALL
Current language:fr_FR

</pre></details><details><summary>Server</summary><pre> 
Operating system: Linux srv-glpi 3.10.0-1160.88.1.el7.x86_64 #1 SMP Tue Mar 7 15:41:52 UTC 2023 x86_64
PHP 8.1.20 apache2handler (Core, PDO, Phar, Reflection, SPL, SimpleXML, Zend OPcache, apache2handler, apcu, bcmath, bz2,
	calendar, ctype, curl, date, dom, exif, fileinfo, filter, ftp, gd, gettext, gmp, hash, iconv, imap, intl, json, ldap, libxml,
	mbstring, mcrypt, mysqli, mysqlnd, openssl, pcre, pdo_mysql, pdo_sqlite, posix, session, shmop, snmp, sockets, sodium, sqlite3,
	standard, sysvmsg, sysvsem, sysvshm, tidy, tokenizer, xml, xmlreader, xmlrpc, xmlwriter, xsl, zip, zlib)
Setup: max_execution_time="600" memory_limit="128M" post_max_size="5M" safe_mode="" session.save_handler="files"
	upload_max_filesize="25M" 
Software: Apache/2.4.6 (CentOS) OpenSSL/1.0.2k-fips PHP/8.1.20 ()
	Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/114.0.0.0 Safari/537.36 Edg/114.0.1823.58
Server Software: MariaDB Server
	Server Version: 10.4.30-MariaDB
	Server SQL Mode: STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION
	Parameters: root@localhost/glpi
	Host info: Localhost via UNIX socket
	
PHP version (8.1.20) is supported.
Sessions configuration is OK.
Allocated memory is sufficient.
mysqli extension is installed.
Following extensions are installed: dom, fileinfo, json, simplexml.
curl extension is installed.
gd extension is installed.
intl extension is installed.
libxml extension is installed.
zlib extension is installed.
The constant SODIUM_CRYPTO_AEAD_XCHACHA20POLY1305_IETF_NPUBBYTES is present.
Database engine version (10.4.30) is supported.
No files from previous GLPI version detected.
The log file has been created successfully.
Write access to /var/www/html/glpi/files/_cache has been validated.
Write access to /var/www/html/glpi/config has been validated.
Write access to /var/www/html/glpi/files/_cron has been validated.
Write access to /var/www/html/glpi/files has been validated.
Write access to /var/www/html/glpi/files/_dumps has been validated.
Write access to /var/www/html/glpi/files/_graphs has been validated.
Write access to /var/www/html/glpi/files/_lock has been validated.
Write access to /var/www/html/glpi/files/_pictures has been validated.
Write access to /var/www/html/glpi/files/_plugins has been validated.
Write access to /var/www/html/glpi/files/_rss has been validated.
Write access to /var/www/html/glpi/files/_sessions has been validated.
Write access to /var/www/html/glpi/files/_tmp has been validated.
Write access to /var/www/html/glpi/files/_uploads has been validated.
For security reasons, SELinux mode should be Enforcing.

Web server root directory configuration is not safe as it permits access to non-public files. See installation documentation for more details.
The following directories should be placed outside "/var/www/html/glpi":
‣ "/var/www/html/glpi/files" ("GLPI_VAR_DIR")
‣ "/var/www/html/glpi/config" ("GLPI_CONFIG_DIR")
You can ignore this suggestion if your web server root directory is "/var/www/html/glpi/public".
Sessions configuration is secured.
OS and PHP are relying on 64 bits integers.
exif extension is installed.
ldap extension is installed.
openssl extension is installed.
Following extensions are installed: bz2, Phar, zip.
Zend OPcache extension is installed.
Following extensions are installed: ctype, iconv, mbstring, sodium.
Write access to /var/www/html/glpi/marketplace has been validated.
Timezones seems loaded in database.

</pre></details><details><summary>GLPI constants</summary><pre> 
GLPI_ROOT: "/var/www/html/glpi"
GLPI_CONFIG_DIR: "/var/www/html/glpi/config"
GLPI_VAR_DIR: "/var/www/html/glpi/files"
GLPI_MARKETPLACE_DIR: "/var/www/html/glpi/marketplace"
GLPI_USE_CSRF_CHECK: "1"
GLPI_CSRF_EXPIRES: "7200"
GLPI_CSRF_MAX_TOKENS: "100"
GLPI_USE_IDOR_CHECK: "1"
GLPI_IDOR_EXPIRES: "7200"
GLPI_ALLOW_IFRAME_IN_RICH_TEXT: false
GLPI_SERVERSIDE_URL_ALLOWLIST: ["/^(https?|feed):\\/\\/[^@:]+(\\/.*)?$/"]
GLPI_TELEMETRY_URI: "https://telemetry.glpi-project.org"
GLPI_INSTALL_MODE: "TARBALL"
GLPI_NETWORK_MAIL: "glpi@teclib.com"
GLPI_NETWORK_SERVICES: "https://services.glpi-network.com"
GLPI_MARKETPLACE_ALLOW_OVERRIDE: true
GLPI_MARKETPLACE_MANUAL_DOWNLOADS: true
GLPI_USER_AGENT_EXTRA_COMMENTS: ""
GLPI_DISABLE_ONLY_FULL_GROUP_BY_SQL_MODE: "1"
GLPI_AJAX_DASHBOARD: "1"
GLPI_CALDAV_IMPORT_STATE: 0
GLPI_DEMO_MODE: "0"
GLPI_CENTRAL_WARNINGS: "1"
GLPI_DOC_DIR: "/var/www/html/glpi/files"
GLPI_CACHE_DIR: "/var/www/html/glpi/files/_cache"
GLPI_CRON_DIR: "/var/www/html/glpi/files/_cron"
GLPI_DUMP_DIR: "/var/www/html/glpi/files/_dumps"
GLPI_GRAPH_DIR: "/var/www/html/glpi/files/_graphs"
GLPI_LOCAL_I18N_DIR: "/var/www/html/glpi/files/_locales"
GLPI_LOCK_DIR: "/var/www/html/glpi/files/_lock"
GLPI_LOG_DIR: "/var/www/html/glpi/files/_log"
GLPI_PICTURE_DIR: "/var/www/html/glpi/files/_pictures"
GLPI_PLUGIN_DOC_DIR: "/var/www/html/glpi/files/_plugins"
GLPI_RSS_DIR: "/var/www/html/glpi/files/_rss"
GLPI_SESSION_DIR: "/var/www/html/glpi/files/_sessions"
GLPI_TMP_DIR: "/var/www/html/glpi/files/_tmp"
GLPI_UPLOAD_DIR: "/var/www/html/glpi/files/_uploads"
GLPI_INVENTORY_DIR: "/var/www/html/glpi/files/_inventories"
GLPI_NETWORK_REGISTRATION_API_URL: "https://services.glpi-network.com/api/registration/"
GLPI_MARKETPLACE_PLUGINS_API_URI: "https://services.glpi-network.com/api/glpi-plugins/"
GLPI_I18N_DIR: "/var/www/html/glpi/locales"
GLPI_VERSION: "10.0.7"
GLPI_SCHEMA_VERSION: "10.0.7@5d45269702917a32805e25b678f6779a98b145f6"
GLPI_MARKETPLACE_PRERELEASES: false
GLPI_MIN_PHP: "7.4.0"
GLPI_MAX_PHP: "8.3.0"
GLPI_YEAR: "2023"

</pre></details><details><summary>Libraries</summary><pre> 
htmlawed/htmlawed version 1.2.9 in (/var/www/html/glpi/vendor/htmlawed/htmlawed)
phpmailer/phpmailer version 6.8.0 in (/var/www/html/glpi/vendor/phpmailer/phpmailer/src)
simplepie/simplepie version 1.5.8 in (/var/www/html/glpi/vendor/simplepie/simplepie/library)
tecnickcom/tcpdf version 6.4.4 in (/var/www/html/glpi/marketplace/pdf/vendor/tecnickcom/tcpdf)
michelf/php-markdown in (/var/www/html/glpi/vendor/michelf/php-markdown/Michelf)
true/punycode in (/var/www/html/glpi/vendor/true/punycode/src)
iamcal/lib_autolink in (/var/www/html/glpi/vendor/iamcal/lib_autolink)
sabre/dav in (/var/www/html/glpi/vendor/sabre/dav/lib/DAV)
sabre/http in (/var/www/html/glpi/vendor/sabre/http/lib)
sabre/uri in (/var/www/html/glpi/vendor/sabre/uri/lib)
sabre/vobject in (/var/www/html/glpi/vendor/sabre/vobject/lib)
laminas/laminas-i18n in (/var/www/html/glpi/vendor/laminas/laminas-i18n/src)
laminas/laminas-servicemanager in (/var/www/html/glpi/vendor/laminas/laminas-servicemanager/src)
monolog/monolog in (/var/www/html/glpi/vendor/monolog/monolog/src/Monolog)
sebastian/diff in (/var/www/html/glpi/vendor/sebastian/diff/src)
donatj/phpuseragentparser in (/var/www/html/glpi/vendor/donatj/phpuseragentparser/src/UserAgent)
elvanto/litemoji in (/var/www/html/glpi/vendor/elvanto/litemoji/src)
symfony/console in (/var/www/html/glpi/vendor/symfony/console)
scssphp/scssphp in (/var/www/html/glpi/vendor/scssphp/scssphp/src)
laminas/laminas-mail in (/var/www/html/glpi/vendor/laminas/laminas-mail/src/Protocol)
laminas/laminas-mime in (/var/www/html/glpi/vendor/laminas/laminas-mime/src)
rlanvin/php-rrule in (/var/www/html/glpi/vendor/rlanvin/php-rrule/src)
blueimp/jquery-file-upload in (/var/www/html/glpi/vendor/blueimp/jquery-file-upload/server/php)
ramsey/uuid in (/var/www/html/glpi/vendor/ramsey/uuid/src)
psr/log in (/var/www/html/glpi/vendor/psr/log/Psr/Log)
psr/simple-cache in (/var/www/html/glpi/vendor/psr/simple-cache/src)
psr/cache in (/var/www/html/glpi/vendor/psr/cache/src)
league/csv in (/var/www/html/glpi/vendor/league/csv/src)
mexitek/phpcolors in (/var/www/html/glpi/vendor/mexitek/phpcolors/src/Mexitek/PHPColors)
guzzlehttp/guzzle in (/var/www/html/glpi/vendor/guzzlehttp/guzzle/src)
guzzlehttp/psr7 in (/var/www/html/glpi/vendor/guzzlehttp/psr7/src)
glpi-project/inventory_format in (/var/www/html/glpi/vendor/glpi-project/inventory_format/lib/php)
wapmorgan/unified-archive in (/var/www/html/glpi/vendor/wapmorgan/unified-archive/src)
paragonie/sodium_compat in (/var/www/html/glpi/vendor/paragonie/sodium_compat/src)
symfony/cache in (/var/www/html/glpi/vendor/symfony/cache)
html2text/html2text in (/var/www/html/glpi/vendor/html2text/html2text/src)
symfony/css-selector in (/var/www/html/glpi/vendor/symfony/css-selector)
symfony/dom-crawler in (/var/www/html/glpi/vendor/symfony/dom-crawler)
twig/twig in (/var/www/html/glpi/vendor/twig/twig/src)
twig/string-extra in (/var/www/html/glpi/vendor/twig/string-extra)
symfony/polyfill-ctype not found
symfony/polyfill-iconv not found
symfony/polyfill-mbstring not found
symfony/polyfill-php80 not found
symfony/polyfill-php81 not found
symfony/polyfill-php82 in (/var/www/html/glpi/vendor/symfony/polyfill-php82)
league/oauth2-client in (/var/www/html/glpi/vendor/league/oauth2-client/src/Provider)
league/oauth2-google in (/var/www/html/glpi/vendor/league/oauth2-google/src/Provider)
thenetworg/oauth2-azure in (/var/www/html/glpi/vendor/thenetworg/oauth2-azure/src/Provider)
phpCas version 1.6.1 in (/usr/share/pear)

</pre></details><details><summary>Plugins list</summary><pre> 
	addressing           Name: Adressage IP                   Version: 3.0.1      State: Enabled                                 
		Install Method: Marketplace
	advancedplanning     Name: advancedplanning               Version: 1.1.0      State: Enabled                                 
		Install Method: Marketplace
	glpiinventory        Name: GLPI Inventory                 Version: 1.2.1      State: Enabled                                 
		Install Method: Marketplace
	pdf                  Name: Impression pdf                 Version: 3.0.0      State: Enabled                                 
		Install Method: Marketplace
	oauthimap            Name: Oauth IMAP                     Version: 1.4.3      State: Enabled                                 
		Install Method: Marketplace

</pre></details>

**Comment by trasher** [comment]
I do not reproduce with latest nightly

**Comment by steph461** [comment]
same issue with last est nightly

```shell
[2023-06-28 15:21:44] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 379
  *** MySQL query error:
  SQL: UPDATE `glpi_agents` SET `name` = 'xxxxxxxxxxxxxxxxxxxx', `items_id` = '' WHERE `id` = '980'
  Error: Incorrect integer value: '' for column `glpi`.`glpi_agents`.`items_id` at row 1
  Backtrace :
  src/DBmysql.php:1468                               
  src/CommonDBTM.php:686                             DBmysql->update()
  src/CommonDBTM.php:1675                            CommonDBTM->updateInDB()
  front/agent.form.php:96                            CommonDBTM->update()
  {"user":"7@srv-glpi"} 
[2023-06-28 15:21:52] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 379
  *** MySQL query error:
  SQL: UPDATE `glpi_agents` SET `name` = 'biomol10', `items_id` = '' WHERE `id` = '980'
  Error: Incorrect integer value: '' for column `glpi`.`glpi_agents`.`items_id` at row 1
  Backtrace :
  src/DBmysql.php:1468                               
  src/CommonDBTM.php:686                             DBmysql->update()
  src/CommonDBTM.php:1675                            CommonDBTM->updateInDB()
  front/agent.form.php:96                            CommonDBTM->update()
  {"user":"7@srv-glpi"}
```

**Comment by steph461** [comment]
[2023-06-28 15:25:47] glpisqllog.ERROR: DBmysql::query() in /var/www/html/glpi/src/DBmysql.php line 379
  *** MySQL query error:
  SQL: UPDATE `glpi_agents` SET `name` = 'test 2', `items_id` = '' WHERE `id` = '980'
  Error: Incorrect integer value: '' for column `glpi`.`glpi_agents`.`items_id` at row 1
  Backtrace :
  src/DBmysql.php:1468                               
  src/CommonDBTM.php:686                             DBmysql->update()
  src/CommonDBTM.php:1675                            CommonDBTM->updateInDB()
  front/agent.form.php:96                            CommonDBTM->update()
  {"user":"7@srv-glpi"}

**Comment by steph461** [comment]
To explain, the computer  has changed NAME. Instead of uninstalling the GLPi agent and reinstalling it to apply the new name, 
I wanted to change its name manually.

To solve this problem, I uninstall the GLPI agent and I reinstall it. If I reinstall without uninstalling, it keeps the old NAME.

**Comment by steph461** [comment]
![image](https://github.com/glpi-project/glpi/assets/68054991/52c7b04f-f87d-41ea-9fbc-d48376a197aa)

**Comment by steph461** [comment]
![image](https://github.com/glpi-project/glpi/assets/68054991/560c1696-8b3e-4940-9140-c406f6def064)

**Comment by trasher** [comment]
OK. I do not reproduce, can't help.

**Comment by cconard96** [comment]
I can recreate the issue. Looks like a simple typo in the twig template prevents the items_id from being present in the hidden input.

Should be fixed by #15053
